### ISSUE TITLE
Disable `reportUnknownArgumentType` if only "know" methods required from argument type

### ISSUE THREAD

**Issue by minmax** [microsoft/pyright#9542]
`pyright` 1.1.389

This code:
```python
def is_tuple_len_1(obj: object) -> bool:
    return isinstance(obj, tuple) and len(obj) == 1
                                          ^^^
```

failed with error:

```
error: Argument type is partially unknown
    Argument corresponds to parameter "obj" in function "len"
    Argument type is "tuple[Unknown, ...]" (reportUnknownArgumentType)
```

And as i can see, `len` don't use any kind of `T` (`Sized` is not subclass of `typing.Generic`).

```python
def len(obj: Sized, /) -> int: ...

class Sized(Protocol, metaclass=ABCMeta):
    @abstractmethod
    def __len__(self) -> int: ...
```

So call `obj.__len__` is fully "known".

It would be useful to exclude such situations from errors.

**Comment by erictraut** [comment]
I don't see a good way to know whether it's safe or unsafe for an `Unknown` to be present. The purpose of this check is to tell you about the condition so you can determine for yourself whether it's going to be a problem. I'm not in favor of hard-coding special cases for functions like `len`. From the perspective of a type checker, this is just like any other function.

For these reasons, I'm going to reject this enhancement request.
